### PR TITLE
Redesign the config file

### PR DESCRIPTION
**Pull request by zhongyi-zhang**
Solve https://github.com/openservicebrokerapi/osb-checker/issues/48.
Also do a refactor on the test framework.

This PR tries to do a minimal change on the test framework to adjust the config redesign. All the test cases are kept. Something that not looks good to me are just commented as `TODO`.
Currently only the mock server can be ensured to pass all the test cases. Will create subsequent issues and PRs to continue improving the checker.

The change in test.js is huge. Git diff works not very well on it. Actually most of the original logic are kept. The refactor only optimizes how to trigger them according to the new config file. Appreciate your reviews!

### CODE REVIEW

File: 2.13/tests/test/configs/config_mock.json [hunk 1]
```diff
@@ -3,106 +3,54 @@
...
+                {
+                    "operation": "unbind"
                 },
-                "previous_values":{
-                    "plan_id":"6ddf6b41-fb60-4b70-af99-8ecc4896b3cf",
-                    "service_id":"2e2fc314-37b6-4587-8127-8f9ee8b33fea",
-                    "organization_id":"org-guid-here",
-                    "space_id":"spcae-guid-here"
+                {
+                    "operation": "deprovision",
```
Comment (leonwanghui) [inline 2.13/tests/test/configs/config_mock.json]: According to the spec, I'm wondering if it's better to add `plan_id` field here.

File: 2.13/tests/test/configs/config_mock.json [hunk 2]
```diff
@@ -3,106 +3,54 @@
...
-    ],
-    "updates":[
-        {
-            "instance_id":"",
-            "body":{
-                "context" :{
-                    "platform": "cloudfoundry",
-                    "some_field": "some-contextual-data"
+                {
+                    "operation": "bind",
```
Comment (leonwanghui) [inline 2.13/tests/test/configs/config_mock.json]: Don't it need some optional fields such like `context` and `bind_resource`?

File: 2.13/tests/test/configs/config_mock.json (hunk 1 above)
Comment (zhongyi-zhang) [reply 2.13/tests/test/configs/config_mock.json]: Let me clarify that, the config file no longer directly specifies the request body. The test framework is able to what the `plan_id` is currently and build the request body.

File: 2.13/tests/test/configs/config_mock.json (hunk 2 above)
Comment (zhongyi-zhang) [reply 2.13/tests/test/configs/config_mock.json]: The field is optional and the test framework doesn't add any case for it. Any time we can add it back for new cases which require it. Does it make sense?

**Comment by duglin** [comment]
LGTM

**Comment by zhongyi-zhang** [comment]
Thanks! Now one more approval is required... @mattmcneeney as you know about the redesign, could you approve it? BTW, can @norshtein, @leonwanghui, and me be added to the .pullapprove.yml? It would help the project move faster on the non-critical PRs (e.g. fixing typo) and engineering-only PRs (e.g. fixing logic which is not reflecting the purpose of a case) -- we are pretty sure that they can be merged.

**Comment by zhongyi-zhang** [comment]
Thanks! Now one more approval is required... @mattmcneeney as you know about the redesign, could you approve it? BTW, can @norshtein, @leonwanghui, and me be added to the .pullapprove.yml? It would help the project move faster on the non-critical PRs (e.g. fixing typo) and engineering-only PRs (e.g. fixing logic which is not reflecting the purpose of a case) -- we are pretty sure that they can be merged.

**Comment by zhongyi-zhang** [comment]
Thanks! Now one more approval is required... @mattmcneeney as you know about the redesign, could you approve it? BTW, can @norshtein, @leonwanghui, and me be added to the .pullapprove.yml? It would help the project move faster on the non-critical PRs (e.g. fixing typo) and engineering-only PRs (e.g. fixing logic which is not reflecting the purpose of a case) -- we are pretty sure that they can be merged.

**Comment by mattmcneeney** [comment]
@zhongyi-zhang I have added @norshtein and @leonwanghui as approvers. Thanks and have fun!

**Comment by mattmcneeney** [comment]
@zhongyi-zhang I have added @norshtein and @leonwanghui as approvers. Thanks and have fun!

**Comment by mattmcneeney** [comment]
@zhongyi-zhang I have added @norshtein and @leonwanghui as approvers. Thanks and have fun!

**Comment by mattmcneeney** [comment]
@zhongyi-zhang I have added @norshtein and @leonwanghui as approvers. Thanks and have fun!

**Comment by mattmcneeney** [comment]
@zhongyi-zhang I have added @norshtein and @leonwanghui as approvers. Thanks and have fun!

**Comment by mattmcneeney** [comment]
@zhongyi-zhang I have added @norshtein and @leonwanghui as approvers. Thanks and have fun!

**Comment by mattmcneeney** [comment]
@zhongyi-zhang I have added @norshtein and @leonwanghui as approvers. Thanks and have fun!

**Comment by norshtein** [comment]
LGTM

**Comment by zhongyi-zhang** [comment]
@mattmcneeney thanks!! Seems the change doesn't have an effect on the existing PR... Just merge it this time.

**Pull request merged**